src/CosmosEditorManager.ts

```diff
@@ -95,7 +95,11 @@ export class CosmosEditorManager {
         const timestamp = (new Date()).toLocaleTimeString();
         ext.outputChannel.appendLine(`${timestamp}: Updated entity "${editor.label}"`);
         ext.outputChannel.show();
-        await this.updateEditor(updatedContent, vscode.window.activeTextEditor, editor);
+        if (doc.isClosed !== true) {
+            const firstRelatedEditor = vscode.window.visibleTextEditors.filter((ed) => ed.document === doc)[0];
```

Comment:
Thank you for the encouragement! Those two points surprised me too when implementing (I started with just a simple if doc.isClosed and then from its documentation I realised the issue of multiple editors for the same document).
1. I am looking for the first matching editor. When updating an editor through the API, VS Code updates the underlying document, thus all visible editors. I tested that with multiple open editors. It's also implied by the [documentation](https://code.visualstudio.com/docs/extensionAPI/vscode-api#TextEditorEdit) of vscode.TextEditorEdit
2. I don't think it could happen, taking into account the single-threaded model of javascript. That said, the condition is what allows me to safely do [0] in the next line. An alternative (but slightly slower) code would be:
```javascript
const relatedEditors = vscode.window.visibleTextEditors.filter((ed) => ed.document === doc);
if (relatedEditors.length > 0) {
//update relatedEditors[0]
```

Reply:
@PrashanthCorp I suppose I'm not supposed to resolve the conversation myself :).

Reply:
> Just curious, is there a case (for example a race condition) where doc.isClosed is false, but doc is not present in visibleTextEditors? The if condition and the filter seem to be checking for the same thing. :)

It's not a race condition, but this is possible. For example, follow these steps:
1. Edit the file, but don't save
1. Open another file (so that it covers the other tab)
1. Select "File -> Save All"

In this case, `isClosed` is false because the editor is still open, but it's not in the list of `visibleTextEditors` because it's behind the other tab.

Please just do `if (firstRelatedEditor) {` before you call `updateEditor`

Reply:
I misunderstood that visibleTextEditors would be all open text editors (it's not clear from the doc)
But in that case, if (firstRelatedEditor) won't do what we want: we want to update the hidden (open) editor with what we got from the server. I don't see a way to get "all open editors" . Anyway, just doing a if (firstRelatedEditor) is already an improvement. I will update the pull request immediately

Reply:
Yes I agree this is an improvement and is worth getting in - even though it's not perfect